examples/chip-tool/commands/common/Command.cpp

```diff
@@ -125,6 +132,17 @@ size_t Command::AddArgument(const char * name, const char * value)
     return mArgs.size();
 }
 
+size_t Command::AddArgument(const char * name, char ** value)
+{
+    Argument arg;
+    arg.type  = ArgumentType::String;
+    arg.name  = name;
+    arg.value = const_cast<void *>(reinterpret_cast<const void *>(value));
```

Comment:
Why not reinterpret_cast to void * ? Going const and then removing the const seems odd.

Reply:
What is the lifetime for these arguments? Maybe for the command tool we should use std::string or some reference counted thing.

Reply:
> Why not reinterpret_cast to void * ? Going const and then removing the const seems odd.

Oops. I will remove that. I blindly copy-paste the the previous method that was expecting a `const char *`

Reply:
> What is the lifetime for these arguments? Maybe for the command tool we should use std::string or some reference counted thing.

Those arguments basically lives for the duration of the program. Those are strings pointed to by `argv`